src/resolvers/ReviewResolver.ts

```diff
@@ -0,0 +1,277 @@
+        const event = await Events.findOne({ where: { id: input.eventId } });
+        if (!event) {
+            throw new Error("Event not found");
+        }
+
+        const campaign = new ReviewCampaign();
+        campaign.title = input.title;
+        campaign.description = input.description || null;
+        campaign.startDate = input.startDate ? new Date(input.startDate) : new Date();
+        campaign.endDate = input.endDate ? new Date(input.endDate) : new Date();
+        campaign.eventId = event;
+        campaign.for = input.for;
+        campaign.for_id = input.for_id || null;
+
+        await campaign.save();
+
+        // Create questions
+        for (const questionInput of input.questions) {
+            const question = new ReviewQuestion();
+            question.question = questionInput.question;
+            question.questionType = questionInput.questionType;
+            question.rangeMin = questionInput.rangeMin || null;
+            question.rangeMax = questionInput.rangeMax || null;
+            question.reviewCampaign = campaign;
+
+            await question.save();
+
+            // Create dropdown options if needed
+            if (questionInput.dropDown && questionInput.questionType === "dropdown") {
+                for (const dropDownInput of questionInput.dropDown) {
+                    const dropDown = new DropDown();
+                    dropDown.text = dropDownInput.text;
+                    dropDown.value = dropDownInput.value;
+                    dropDown.reviewQuestion = question;
+                    await dropDown.save();
+                }
+            }
+        }
+
+        const result = await ReviewCampaign.findOne({
+            where: { id: campaign.id },
+            relations: ["questions", "questions.dropDown"]
+        });
+
+        return { reviewCampaign: result };
+    }
+
+    @Mutation(() => ReviewCampaign)
+    @UseMiddleware(isAuth)
+    async updateReviewCampaign(@Arg("input") input: UpdateReviewCampaignInput) {
+        const campaign = await ReviewCampaign.findOne({ where: { id: input.id } });
+        if (!campaign) {
+            throw new Error("Review campaign not found");
+        }
+
+        if (input.title !== undefined) campaign.title = input.title;
+        if (input.description !== undefined) campaign.description = input.description;
+        if (input.startDate !== undefined) campaign.startDate = input.startDate ? new Date(input.startDate) : null;
+        if (input.endDate !== undefined) campaign.endDate = input.endDate ? new Date(input.endDate) : null;
+        if (input.for !== undefined) campaign.for = input.for;
+        if (input.for_id !== undefined) campaign.for_id = input.for_id;
+
+        await campaign.save();
+        return campaign;
+    }
+
+    @Mutation(() => Boolean)
+    @UseMiddleware(isAuth)
+    async deleteReviewCampaign(@Arg("id") id: string) {
```

Comment:
Before you can send review_campaign you need to verify user is logged in and authenticated. And your sending them the right review, and not sending them the wrong thing